chia/cmds/coin_funcs.py

```diff
@@ -176,7 +176,7 @@ async def async_combine(args: Dict[str, Any], wallet_client: WalletRpcClient, fi
 async def async_split(args: Dict[str, Any], wallet_client: WalletRpcClient, fingerprint: int) -> None:
     wallet_id: int = args["id"]
     number_of_coins = args["number_of_coins"]
-    fee = Decimal(args["fee"])
+    final_fee = uint64(int(Decimal(args["fee"] * units["chia"])))
```

Comment:
what's the point of creating a `Decimal` object and an `int` object here? `args` is presumably some `int`-like object since you can multiply it. So, after the multiplication, I would expect we would already have `int` and could just pass it directly to `uint64()`. What am I missing?

Reply:
args is a float / decimal because we express it in xch, once we convert it to chia thats not the case and its always an int.

Reply:
is it not the same to say this:
```
 final_fee = uint64(args["fee"] * units["chia"])
```
?

Reply:
mypy complains if you do that, we use decimal because args["fee"] is a str

Reply:
in that case, I'm pretty sure that multiplication is a bug

Reply:
> in that case, I'm pretty sure that multiplication is a bug

?

Reply:
you want to convert the string to a number and then multiply it, right?
What this line does currently is to "multiply" the string, which will duplicate it. You may end up with a string containing multiple decimal points. I would hope `Decimal()` throws if it's passed such string though